src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -126,6 +124,36 @@ create_distributed_table(PG_FUNCTION_ARGS)
 
 	char *distributionColumnName = text_to_cstring(distributionColumnText);
 	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
+	text *colocateWithTableNameText = NULL;
+	char *colocateWithTableName = NULL;
+
+	/* guard against a binary update without a function update */
```

Comment:
Yep, we should generally perform such checks, because it is not always easy for administrators to perform `ALTER EXTENSION` right after an upgrade and in some cases the `ALTER EXTENSION` might not even succeed (e.g. someone manually created a function with the same signature). Citus cloud currently upgrades Citus by building a new AMI and then replicating the data from an existing node into a new node with the new AMI. It's not clear at what point the `ALTER EXTENSION` should happen in that case or what to do if it fails. Ideally, the system can function normally for prolonged periods without the `ALTER EXTENSION` happening.